pkgs/tools/filesystems/blobfuse/default.nix

```
@@ -1,19 +1,40 @@
-{ lib, stdenv, fetchFromGitHub, cmake, pkg-config, curl, gnutls, libgcrypt, libuuid, fuse }:
+{ lib, stdenv, fetchFromGitHub, cmake, pkg-config, curl, gnutls, libgcrypt, libuuid, fuse, boost }:
 
-stdenv.mkDerivation rec {
-  pname = "blobfuse";
-  version = "1.0.2";
-
-  src = fetchFromGitHub {
-    owner  = "Azure";
-    repo   = "azure-storage-fuse";
-    rev    = "v${version}";
-    sha256 = "1qh04z1fsj1l6l12sz9yl2sy9hwlrnzac54hwrr7wvsgv90n9gbp";
+let
+src_version = "1.3.7";
+src_name = "blobfuse";
+sources = fetchFromGitHub {
+  owner  = "Azure";
+  repo   = "azure-storage-fuse";
+  rev    = "${src_name}-${src_version}-Linux";
+  sha256 = "sha256-yihIuS4AG489U7eBi/p7H6S7Cg54kkQeNVCexxQZ60A=";
+};
+cpplite =
+  stdenv.mkDerivation rec {
+    pname = "cpplite";
+    version = src_version;
+
+    src = sources;
+
+    sourceRoot = "source/cpplite";
+    patches = [ ./install-adls.patch ];
+
+    cmakeFlags = [ "-DBUILD_ADLS=ON" "-DUSE_OPENSSL=OFF" ];
+
+    buildInputs = [ curl libuuid gnutls ];
+    nativeBuildInputs = [ cmake pkg-config ];
   };
+in
+
+stdenv.mkDerivation rec {
+  pname = src_name;
+  version = src_version;
+
+  src = sources;
```

This is just my preference, don't feel strongly about it though
```suggestion
let
  version = "1.3.7";
  src = fetchFromGitHub {
   owner  = "Azure";
    repo   = "azure-storage-fuse";
    rev    = "blobfuse-${src_version}-Linux";
    sha256 = "sha256-yihIuS4AG489U7eBi/p7H6S7Cg54kkQeNVCexxQZ60A=";
  };
  cpplite =
    stdenv.mkDerivation rec {
      pname = "cpplite";
      inherit version src;

      sourceRoot = "source/cpplite";
      patches = [ ./install-adls.patch ];

      cmakeFlags = [ "-DBUILD_ADLS=ON" "-DUSE_OPENSSL=OFF" ];

      buildInputs = [ curl libuuid gnutls ];
      nativeBuildInputs = [ cmake pkg-config ];
   };
in stdenv.mkDerivation rec {
  pname = "blobfuse";
  inherit version src;
```

---

Thanks a lot for the hints, much appreciated. I'd say that your version is better. New PR incoming, rebased to current master and squashed commits.